frame/plasm-rewards/src/lib.rs

```diff
@@ -0,0 +1,428 @@
+            }
+        }
+    }
+
+    /// End a session potentially ending an era.
+    fn end_session(session_index: SessionIndex) {
+        if let Some(active_era) = Self::active_era() {
+            if let Some(next_active_era_start_session_index) =
+                Self::eras_start_session_index(active_era.index + 1)
+            {
+                if next_active_era_start_session_index == session_index + 1 {
+                    Self::end_era(active_era, session_index);
+                }
+            }
+        }
+    }
+
+    /// * Increment `active_era.index`,
+    /// * reset `active_era.start`,
+    /// * update `BondedEras` and apply slashes.
+    fn start_era(start_session: SessionIndex) {
+        let active_era = <ActiveEra<T>>::mutate(|active_era| {
+            let new_index = active_era.as_ref().map(|info| info.index + 1).unwrap_or(0);
+            *active_era = Some(ActiveEraInfo {
+                index: new_index,
+                // Set new active era start in next `on_finalize`. To guarantee usage of `Time`
+                start: None,
+            });
+            new_index
+        });
+
+        // let bonding_duration = T::BondingDuration::get();
+
+        BondedEras::mutate(|bonded| {
+            bonded.push((active_era, start_session));
+
+            // if active_era > bonding_duration {
+            //     let first_kept = active_era - bonding_duration;
+            //
+            //     // prune out everything that's from before the first-kept index.
+            //     let n_to_prune = bonded.iter()
+            //         .take_while(|&&(era_idx, _)| era_idx < first_kept)
+            //         .count();
+            //
+            //     // kill slashing metadata.
+            //     for (pruned_era, _) in bonded.drain(..n_to_prune) {
+            //         slashing::clear_era_metadata::<T>(pruned_era);
+            //     }
+            //
+            //     if let Some(&(_, first_session)) = bonded.first() {
+            //         T::SessionInterface::prune_historical_up_to(first_session);
+            //     }
+            // }
+        });
+    }
+
+    /// Compute payout for era.
+    fn end_era(active_era: ActiveEraInfo<MomentOf<T>>, _session_index: SessionIndex) {
+        // Note: active_era_start can be None if end era is called during genesis config.
+        if let Some(active_era_start) = active_era.start {
+            // The set of total amount of staking.
+            let now = T::Time::now();
+            let era_duration = now - active_era_start;
+
+            if !era_duration.is_zero() {
+                let total_payout = T::Currency::total_issuance();
+                let for_dapps = T::GetForDappsStaking::get_era_staking_amount(active_era.index);
+                let for_security =
+                    T::GetForSecurityStaking::get_era_staking_amount(active_era.index);
+                println!("for_security:{:?}, for_dapps:{:?}", for_security, for_dapps);
```

Comment:
`println!` instead of `support::print`

Reply:
Sorry, this's debugging comment. I forget to delete it. 🙇